src/layout/TagsInput.js

```diff
@@ -8,9 +8,23 @@ import { Flex } from './Flex'
 import Popover from './Popover'
 import OutsideClickHandler from 'react-outside-click-handler'
 
-let isValidTag = (tag, tags) => {
-  let cleanTag = _.trim(tag)
-  return !_.isEmpty(cleanTag) && !_.includes(cleanTag, tags)
+let isValidInput = (tag, tags) => !_.isEmpty(tag) && !_.includes(tag, tags)
+
+// extracts only the unique (and not already used tags) from the user tags input
+// e.g. if "x,y,z" are already entered then only "d" would be added if the user tris to enter "x,y,d"
+// returns a string with either single tag or multiple unique tags separated by comma
+let getUniqueTagsStringFromInput = (tag, tags, useCommaSeparator = true) => {
+  let trimmedTag = _.trim(tag)
+  if (useCommaSeparator && _.includes(',', trimmedTag)) {
+    // Extract only the unique (and not present in the current tags collection) tags from the string
+    let newUniqTags = _.difference(
+      _.uniq(_.compact(_.split(',', trimmedTag))),
```

Comment:
Re. your comment about `flow` breaking with `difference`, this seems to work (though I only tested it briefly, so I might have missed something):

```es
_.flow(
  _.split(','),
  _.compact,
  _.uniq,
  _.difference(_, tags)
)(trimmedTag)
```

or you could keep `difference` outside and just flow `split`, `compact` and `uniq`:

```es
_.difference(
  _.flow(
    _.split(','),
    _.compact,
    _.uniq,
  )(trimmedTag)
)(tags)
```

I personally think these are cleaner than the nested calls, but it's up to you 🙂

Reply:
Right.

I usually would go with `flow` but in this case this is not as straight forward to follow with the `_.difference(_, tags)` or with the `_difference` and then the inside `_.flow`. In this case I prefer the nested approach.